day-trader-java-app/daytrader-gateway/src/main/java/com/ofss/daytrader/gateway/controller/GatewayController.java

```diff
@@ -737,7 +761,152 @@
+					new BigDecimal(-1), new BigDecimal(-1), 1.0));
+			topLosersData.add(new QuoteDataBean("s:707", "Company707", 8.00, new BigDecimal(196), new BigDecimal(196),
+					new BigDecimal(-1), new BigDecimal(-1), 1.0));
+			topLosersData.add(new QuoteDataBean("s:708", "Company708", 7.00, new BigDecimal(197), new BigDecimal(197),
+					new BigDecimal(-1), new BigDecimal(-1), 1.0));
+			topLosersData.add(new QuoteDataBean("s:709", "Company709", 6.00, new BigDecimal(198), new BigDecimal(198),
+					new BigDecimal(-1), new BigDecimal(-1), 1.0));
+			topLosersData.add(new QuoteDataBean("s:710", "Company710", 5.00, new BigDecimal(199), new BigDecimal(199),
+					new BigDecimal(-1), new BigDecimal(-1), 1.0));
+
+			MarketSummaryDataBean marketSummaryData = new MarketSummaryDataBean(TSIA, openTSIA, volume, topGainersData,
+					topLosersData);
+
+			System.out.println("marketSummaryData from fall back method" + marketSummaryData);
+			// return marketSummaryData;
+			return new ResponseEntity<MarketSummaryDataBean>(marketSummaryData, getNoCacheHeaders(), HttpStatus.OK);
+		}
+	}
+	@SuppressWarnings("unchecked")
+	public ResponseEntity<Collection<HoldingDataBean>> getHoldingsFallback(@PathVariable("userId") String userId) {
+		Log.traceEnter("getHoldingsFallback of GatewayController");
+
+		Collection<HoldingDataBean> holdings = new ArrayList<HoldingDataBean>(3);
+		String userKey = "holdings_" + userId;
+
+		if (CachedObjectsClass.getInstance().checkCacheForObject(userKey) != null) {
+			System.out.println("data is displayed from cache");
+			HoldingDataBean holdingDataBean = (HoldingDataBean) CachedObjectsClass.getInstance()
+					.checkCacheForObject(userKey);
+			return new ResponseEntity<Collection<HoldingDataBean>>((Collection<HoldingDataBean>) holdingDataBean,
+					getNoCacheHeaders(), HttpStatus.OK);
+
+		} else {
+
+			holdings.add(new HoldingDataBean(701, 107d, new BigDecimal(10), new Timestamp(System.currentTimeMillis()),
+					"s:701"));
+			holdings.add(new HoldingDataBean(702, 17d, new BigDecimal(10), new Timestamp(System.currentTimeMillis()),
+					"s:702"));
+			holdings.add(new HoldingDataBean(703, 177d, new BigDecimal(10), new Timestamp(System.currentTimeMillis()),
+					"s:703"));
+			return new ResponseEntity<Collection<HoldingDataBean>>(holdings, getNoCacheHeaders(), HttpStatus.OK);
+		}
+	}
+
+	public ResponseEntity<QuoteDataBean> getQuoteFallback(@PathVariable("symbol") String symbol) {
+		String quotekey = "getQuote_" + symbol;
+
+		if (CachedObjectsClass.getInstance().checkCacheForObject(quotekey) != null) {
+			System.out.println("data is displayed from cache");
+			QuoteDataBean quoteDataBean = (QuoteDataBean) CachedObjectsClass.getInstance()
+					.checkCacheForObject(quotekey);
+			return new ResponseEntity<QuoteDataBean>(quoteDataBean, getNoCacheHeaders(), HttpStatus.OK);
+
+		} else {
+			QuoteDataBean quoteData = new QuoteDataBean("s:701", "Company701", 9.00, new BigDecimal(195),
```

Comment:
It should not be s:701.  it should be the same value of input parameter

Reply:
changes are done as mentioned